From 6fb1a7245a2ff2a98bc7aae6c75522b1e447e1f9 Mon Sep 17 00:00:00 2001
From: Zuohai Xu <zouhaix@marvell.com>
Date: Tue, 11 Jun 2019 16:37:10 +0800
Subject: [PATCH 1170/1200] arm: dts: Update dtsi of cn9130 crb

1. Update GPIO pins for SD VCC and VCQ
2. Align mpps configuration

Change-Id: Ie5c531c3dc9c27302f3f973e4d15cae986fed1d9
Signed-off-by: Zuohai Xu <zouhaix@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/10714
Reviewed-by: Kostya Porotchkin <Kostya.Porotchkin@cavium.com>
Reviewed-by: Stefan Chulski <Stefan.Chulski@cavium.com>
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
---
 arch/arm/dts/cn9130-crb.dtsi | 56 +++++++++++++++++++++---------------
 1 file changed, 33 insertions(+), 23 deletions(-)

diff --git a/arch/arm/dts/cn9130-crb.dtsi b/arch/arm/dts/cn9130-crb.dtsi
index 2b82bbd883..c892811c79 100644
--- a/arch/arm/dts/cn9130-crb.dtsi
+++ b/arch/arm/dts/cn9130-crb.dtsi
@@ -39,7 +39,7 @@
 				regulator-name = "cp0_sd_vccq";
 				regulator-min-microvolt = <1800000>;
 				regulator-max-microvolt = <3300000>;
-				gpios = <&cp0_gpio0 25 GPIO_ACTIVE_HIGH>;
+				gpios = <&cp0_gpio1 18 GPIO_ACTIVE_HIGH>;
 				states = <1800000 0x1
 					  3300000 0x0>;
 			};
@@ -48,7 +48,7 @@
 				regulator-name = "cp0_sd_vcc";
 				regulator-min-microvolt = <3300000>;
 				regulator-max-microvolt = <3300000>;
-				gpio = <&cp0_gpio1 19 GPIO_ACTIVE_HIGH>;
+				gpio = <&cp0_gpio1 22 GPIO_ACTIVE_HIGH>;
 				enable-active-high;
 				regulator-always-on;
 			};
@@ -87,34 +87,44 @@
 &cp0_pinctl {
 	/* MPP Bus:
 	 *	[0-11]	RGMII1
-	 *	[12-16]	SPI1
-	 *	[17-28] GPIO
-	 *	[29-30] UART0
-	 *	[31-32] GPIO
-	 *	[33]	SDIO
+	 *	[12-26]	GPIO
+	 *	[27-30]	SPI1
+	 *	[31]	GPIO
+	 *	[32]	GPIO
+	 *	[33]	SD_PWR_OFF
 	 *	[34]	CP_PCIE0_CLKREQn
-	 *	[35-36]	I2C1
-	 *	[37-38]	I2C0
+	 *	[35-38]	I2C1 I2C0
 	 *	[39]	GPIO
-	 *	[40-41]	UART1
-	 *	[42-43] SD Control
-	 *	[44]	GPIO
-	 *	[45]	CP_PCIE_RSTOUTn
-	 *	[46]	GPIO
-	 *	[47-50] SMI/XSMI
-	 *	[51-55] GPIO
-	 *	[56-62]	SDIO
+	 *	[40-43]	SMI/XSMI
+	 *	[44-46]	GPIO
+	 *	[47]	UART1_TX
+	 *	[48]	GPIO
+	 *	[49]	SD_HST_18_EN
+	 *	[50]	GPIO
+	 *	[51]	SD_PWR_0
+	 *	[52]	PCIE_RSTn
+	 *	[53]	UART1_RX
+	 *	[54]	GPIO
+	 *	[55]	SD_DT
+	 *	[56-61]	SDIO
 	 */
 		/*   0   1   2   3   4   5   6   7   8   9 */
 	pin-func = < 3   3   3   3   3   3   3   3   3   3
-		     3   3   3   3   3   3   3   0   0   0
-		     0   0   0   0   0   0   0   0   0   0xa
-		     0xa 0   0   6   9   2   2   2   2   0
-		     7   7   1   1   0   9   0   8   8   8
-		     8   0   0   0   0   0   0xe 0xe 0xe 0xe
+		     3   3   0   0   0   0   0   0   0   0
+		     0   0   0   0   0   0   0   2   2   2
+		     2   0   0   6   9   2   2   2   2   0
+		     8   8   8   8   0   0   0   7   0   0xa
+		     0   0xa 9   7   0   0xb 0xe 0xe 0xe 0xe
 		     0xe 0xe 0xe>;
 };
 
+&cp0_pinctl {
+	cp0_spi1_pins_crb: cp0-spi-pins-crb {
+		marvell,pins = < 27 28 29 30 >;
+		marvell,function = <2>;
+	};
+};
+
 /*
  * CP0
  */
@@ -138,7 +148,7 @@
 
 &cp0_spi1 {
 	pinctrl-names = "default";
-	pinctrl-0 = <&cp0_spi0_pins>;
+	pinctrl-0 = <&cp0_spi1_pins_crb>;
 	reg = <0x700680 0x50>,		/* control */
 	      <0x2000000 0x1000000>,	/* CS0 */
 	      <0 0xffffffff>,		/* CS1 */
-- 
2.22.0

